### PR TITLE
Udp scan in client

### PR DESCRIPTION
**Pull request by mir3x**
Removed main loop in scan to signal/slot so thread could be removed there.

### CODE REVIEW

File: client/servers.cpp [hunk 1]
```diff
@@ -873,27 +667,14 @@ void server_scan_finish(struct server_scan *scan)
...
     fc_destroy_mutex(&scan->meta.mutex);
 
-    /* This mainly duplicates code from below "else" block.
-     * That's intentional, since they will be completely different in future
-     * versions. We are better prepared for that by having them separately
-     * already. */
-    if (scan->sock >= 0) {
-      fc_closesocket(scan->sock);
-      scan->sock = -1;
-    }
```
Comment (lmoureaux) [inline client/servers.cpp:883]: Missing `fcUdpScan::i()->drop();`?

File: client/servers.cpp [hunk 2]
```diff
@@ -15,70 +15,29 @@
...
-#ifdef HAVE_SYS_UIO_H
-#include <sys/uio.h>
-#endif
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
-
 // Qt
 #include <QByteArray>
+#include <QDebug>
```
Comment (lmoureaux) [inline client/servers.cpp:20]: Unused? You got rid of my QtDebug (oops) but it came back :D

File: client/servers.cpp (hunk 1 above)
Comment (mir3x) [reply client/servers.cpp:883]: Its there in server scan finish line 677 ( no idea how to link code :D )
`  } else {
    fcUdpScan::i()->drop();`

File: client/servers.cpp [hunk 3]
```diff
@@ -873,27 +667,14 @@ void server_scan_finish(struct server_scan *scan)
...
       scan->srvrs.servers = NULL;
       fc_release_mutex(&scan->srvrs.mutex);
     }
   } else {
-    if (scan->sock >= 0) {
-      fc_closesocket(scan->sock);
-      scan->sock = -1;
-    }
-
+    fcUdpScan::i()->drop();
```
Comment (mir3x) [inline client/servers.cpp:677]: Its here ( if  not global scan)

Comment (lmoureaux) [reply client/servers.cpp:677]: Oh I see, it's only used for local scans

**Pull request merged**